src/test/util/CMakeLists.txt

```diff
@@ -17,22 +17,13 @@ add_library(test_util STATIC EXCLUDE_FROM_ALL
   transaction_utils.cpp
   txmempool.cpp
   validation.cpp
+  $<$<BOOL:${ENABLE_WALLET}>:${PROJECT_SOURCE_DIR}/src/wallet/test/util.cpp>
 )
+
 target_link_libraries(test_util
   PRIVATE
     core
-    bitcoin_common
-    bitcoin_node
-    leveldb
-    univalue
     Boost::headers
+  PUBLIC
+    univalue
```

Comment:
I was going to comment that I'd rather see this be `PRIVATE` and have univalue included by the binaries as needed, but test_util pretty clearly exports univalue as part of its interface, so `PUBLIC` univalue actually follows the code as-is quite well I think.

Reply:
Placing `univalue` into usage requirements, and removing it from build specifications for the `bench_bitcoin` and `test_bitcoin` targets, fixes actually, "ld: warning: ignoring duplicate libraries" on macOS.